Sources/SwiftDocC/Model/Rendering/References/CallToActionReference.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A specialized ``DownloadReference`` used for ``CallToAction`` directives.
+///
+/// `@CallToAction` directives can link either to a local file or to a URL, whether relative or
+/// absolute. Directives that use the `file` argument will create a ``DownloadReference`` and copy
+/// the file from the catalog into the resulting archive.
+///
+/// A `CallToActionReference` is intended to encode to Render JSON compatible with a
+/// ``DownloadReference``, but with the `url` set to the text given in the `@CallToAction`'s `url`
+/// argument.
+public struct CallToActionReference: RenderReference, URLReference {
+    public static var baseURL: URL = .init(string: "/\(DownloadReference.locationName)/")!
+
+    public var type: RenderReferenceType = .download
+
+    public var identifier: RenderReferenceIdentifier
+
+    enum CodingKeys: String, CodingKey {
+        case type
+        case identifier
+        case url
+    }
+
+    public init(identifier: RenderReferenceIdentifier) {
+        self.identifier = identifier
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.identifier = try container.decode(RenderReferenceIdentifier.self, forKey: .identifier)
```

Comment:
Will this round-trip correctly?

Reply:
If you go `CallToActionReference -> DownloadReference -> DownloadReference`, the second round-trip will come back differently, since the URL will be different. However, if you go `CallToActionReference -> CallToActionReference` or `CallToActionReference -> DownloadReference -> CallToActionReference`, then the information should be preserved. (The second one will have a different JSON encoding when going back to `CallToActionReference`, but the final decoded instance should be the same as the original one, since the only valuable piece of information is the identifier, which remains untouched.)

Should i add more tests to this effect?

Reply:
I think that would be good. This is tricky behavior that seems like it would be easy to break – having a test to enforce it will likely help us down the road.